pkg/asset/manifests/dns.go

```diff
@@ -85,7 +84,7 @@ func (d *DNS) Generate(dependencies asset.Parents) error {
 			if err != nil {
 				return errors.Wrapf(err, "getting public zone for %q", installConfig.Config.BaseDomain)
 			}
-			config.Spec.PublicZone = &configv1.DNSZone{ID: strings.TrimPrefix(*zone.Id, "/hostedzone/")}
+			config.Spec.PublicZone = &configv1.DNSZone{ID: *zone.Id}
```

Comment:
i think the function is missing the handling of empty publiczoneid by the user in install config.

Reply:
Fixed.